msal/src/main/java/com/microsoft/identity/client/AuthenticationActivity.java

```diff
@@ -288,4 +240,42 @@ private void sendError(final String errorCode, final String errorDescription) {
         errorIntent.putExtra(Constants.UIResponse.ERROR_DESCRIPTION, errorDescription);
         returnToCaller(Constants.UIResponse.AUTH_CODE_ERROR, errorIntent);
     }
-}
+
+    private MicrosoftStsAuthorizationRequest getAuthorizationRequestFromIntent(final Intent callingIntent) {
```

Comment:
Do we have to use this specific return type? It looks like we _may_ be able to get away with casting-to and returning the supertype `AuthorizationRequest` -- you'll have to make the generic type params of `mEmbeddedWebViewAuthorizationStrategy` use the supertype too. Ideally, we keep this `Activity`'s implementation as non-Microsoft specific as possible

It looks like all we do is call `requestAuthorization()`, right?

Reply:
Thank you for the comments Brian! I got some confusions on this as well as the implementation of CoreAdapter. Please let me know when you have available to discuss about this offline.